src/IO/ReadWriteBufferFromHTTP.h

```diff
@@ -176,20 +191,21 @@ namespace detail
 
             LOG_TRACE(log, "Sending request to {}", uri_.toString());
 
-            auto sess = session->getSession();
-
+            auto sess = current_session->getSession();
             try
             {
                 auto & stream_out = sess->sendRequest(request);
 
                 if (out_stream_callback)
                     out_stream_callback(stream_out);
 
-                istr = receiveResponse(*sess, request, response, true);
+                auto result_istr = receiveResponse(*sess, request, response, true);
                 response.getCookies(cookies);
 
-                content_encoding = response.get("Content-Encoding", "");
-                return istr;
+                if constexpr (!use_initial_range)
+                    content_encoding = response.get("Content-Encoding", "");
```

Comment:
not obvious why we need to add check for `use_initial_range` here, add a comment?

Reply:
what do you think about renaming it to `for_head_request` or `for_object_info`?
because using `initial_range` in other cases (or running any request in parallel using the same buffer) is really sketchy

Reply:
yeah I like this idea